astroid/transforms.py

```diff
@@ -4,9 +4,35 @@
 
 import collections
 from functools import lru_cache
+from typing import TYPE_CHECKING
 
+if TYPE_CHECKING:
+    from astroid import NodeNG
 from astroid.context import _invalidate_cache
 
+TRANSFORM_MAX_CACHE_SIZE = 10000
+
+
+@lru_cache(maxsize=TRANSFORM_MAX_CACHE_SIZE)
```

Comment:
What if we just made no changes in this part but for removing caching altogether?

I added a print statement on the existing code on main and saw no hits (the print statement won't execute if there were a hit, but each print on a miss would ideally show some hits!):
```diff
diff --git a/astroid/transforms.py b/astroid/transforms.py
index f74203d0..7d6b1cc3 100644
--- a/astroid/transforms.py
+++ b/astroid/transforms.py
@@ -44,6 +44,7 @@ class TransformVisitor:
                 if ret.__class__ != cls:
                     # Can no longer apply the rest of the transforms.
                     break
+        print(self._transform.cache_info())
         return node
 
     def _visit(self, node):
```
```
...
CacheInfo(hits=0, misses=435675, maxsize=10000, currsize=10000)
```
Am I missing something? 🤔

Reply:
😅 

I wonder why we're missing these. The signature seems rather simple?

Reply:
I linted `pylint` itself in my example. I guess there is no need to transform a certain node with a certain transform twice. (Makes sense!)

Reply:
It's also not causing any CI failures as is. But probably making the code slower because of the overhead of the LRU tracking.